azure_functions_worker/bindings/meta.py

```diff
@@ -300,6 +300,15 @@ def deferred_bindings_decode(binding: typing.Any,
     """
     global deferred_bindings_cache
 
+    # Only applies to Event Hub and Service Bus - cannot cache
+    # These types will always produce different content and are not clients
+    if (datum.type == "collection_model_binding_data"
+            or datum.value.source == "AzureEventHubsEventData"
+            or datum.value.source == "AzureServiceBusReceivedMessage"):
+        return binding.decode(datum,
+                              trigger_metadata=metadata,
+                              pytype=pytype)
+
```

Comment:
NIT: Improvement suggestion: 

validSources = {
    "AzureEventHubsEventData",
    "AzureServiceBusReceivedMessage"
}

if datum.type == "collection_model_binding_data" or datum.value.source in valid_sources:
    return binding.decode(datum, trigger_metadata=metadata, pytype=pytype)
    
 Also, can type can be enum in your datum model? That will avoid the string comparisons and improve the codebae as less strings constant will flow.